### PR TITLE
Consecutive initial marks/dashes spacing tweak + some sync with Upstream

### PR DESCRIPTION
**Pull request by poire-z**
#### (Upstream) Fix rescaled book cover on 4bpp screen
https://github.com/buggins/coolreader/commit/43a4feb67861a51cf9dcbba060b572a877a47104

#### (Upstream) Page splitting: fix possible infinite loop

https://github.com/buggins/coolreader/commit/d69818193bc883a6852ec040b6f45255e4b3b329
Can't trigger it on KOReader.
Quick fix (should probably be ensured earlier) picked just to not diverge.

#### (Upstream) Correct the language code for Greek

https://github.com/buggins/coolreader/commit/f1d309684006f6d0479bc32baad83a7e5a0847de
Used with text files when detecting encoding.

#### (Upstream) Fix typo in LVDocView::propsApply()

https://github.com/buggins/coolreader/commit/38453366b5821df44fb7028c4cd5ec10f6762698

#### (Upstream) Large files support (engine part)

https://github.com/buggins/coolreader/commit/29707a0f9abf91179c6c5873e37f930bcd49eb05
Mostly large zip-files. Integer overflow fixed.

#### (Upstream) Make crengine.font.fallback.faces plural

https://github.com/buggins/coolreader/commit/e7c85ef565bd5c5781b0e24b3936a17541dbd7c6
Upstream had been keeping both SetFallbackFontFace() and SetFallbackFontFaces(), and recently removed the former.  We've been having only the plural, but kept the setting singular for compatibility.
Make everything plural to keep in sync.

#### (Upstream) Option to not limit font size to a set

https://github.com/buggins/coolreader/commit/3a2bc0142cff87a4784df58618c216ab909f682d
Previously, the default font size was limited to a (configurable) set of sizes.
This adds an option to not enforce that, and use the requested font size as-is.
(Set USE_LIMITED_FONT_SIZES_SET=0 for KOReader, as cre.cpp needs the correct headers - and it feels awkward to pollute our base/Makefile with such a little -D.)

#### Text: dont adjust space after consecutive initial marks/dashes

See https://github.com/koreader/koreader/issues/7366#issuecomment-801890487.
Extend 2504a403 to keep doing it when multiple such initial quotation marks/dashes separated by spaces.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/crengine/430)

### CODE REVIEW

**Comment by NiLuJe** [comment]
Whee, that mans we'll be able to get rid of that stupid font sizes array in cre.cpp \o/.

File: crengine/Tools/langstat/langstat.cpp [hunk 1]
```diff
@@ -63,8 +63,8 @@ void MakeStats()
    MakeStatsForFile( BASE_DIR "es2-utf8.txt", "utf8", "ee", 2, f, list );
    MakeStatsForFile( BASE_DIR "ee3-cp775.txt", "cp775", "ee", 1, f, list );
    MakeStatsForFile( BASE_DIR "es3-cp1257.txt", "cp1257", "ee", 2, f, list );
-   MakeStatsForFile( BASE_DIR "gr1-cp1253.txt", "cp1253", "gr", 1, f, list );
-   MakeStatsForFile( BASE_DIR "gr1-cp737.txt", "cp737", "gr", 1, f, list );
+   MakeStatsForFile( BASE_DIR "gr1-cp1253.txt", "cp1253", "el", 1, f, list );
```
Comment (Frenzie) [inline crengine/Tools/langstat/langstat.cpp:66]: Strange, didn't we discuss this on our end recently?

Comment (NiLuJe) [reply crengine/Tools/langstat/langstat.cpp:66]: Now that you mention it, that does vaguely ring a bell (perhaps not strictly in the same context, though)...

Comment (poire-z) [reply crengine/Tools/langstat/langstat.cpp:66]: Probably in some hyphenation topic, like https://github.com/koreader/crengine/issues/373#issuecomment-747671695.

**Pull request merged**